legistar/bills.py

```diff
@@ -450,17 +452,30 @@ def text(self, matter_id, latest_version_value=None):
     def legislation_detail_url(self, matter_id):
         gateway_url = self.BASE_WEB_URL + '/gateway.aspx?m=l&id={0}'
 
-        # we want to supress any session level params for this head request
+        # We want to supress any session level params for this head request,
         # since they could lead to an additonal level of redirect.
         #
         # Per
         # http://docs.python-requests.org/en/master/user/advanced/, we
         # have to do this by setting session level params to None
-        legislation_detail_route = self.head(
+        response = self.head(
             gateway_url.format(matter_id),
-            params={k: None for k in self.params}).headers['Location']
-
-        return urljoin(self.BASE_WEB_URL, legislation_detail_route)
+            params={k: None for k in self.params}
+        )
+
+        # If the gateway URL redirects, the matter is publicly viewable. Grab
+        # its detail URL from the response headers.
+        if response.status_code == 302:
+            legislation_detail_route = response.headers['Location']
+            return urljoin(self.BASE_WEB_URL, legislation_detail_route)
+
+        # If the status code is anything but a 200 or 302, something is wrong.
+        # Raise an HTTPError to interrupt the scrape.
+        elif response.status_code != 200:
```

Comment:
let's go ahead and change the status code to 503 here, instead of in _check_errors. Let's also log out what the actual code was.